File: Orc.cpp
#include "Orc.h"


Orc::Orc() :Entity()
{
	club = 0;
	CoinDrop = 0;
	InstType = OrcType;
}

Orc::Orc(int club, int CoinDrop, int health) :Entity(health)
{
	this->club = club;
	this->CoinDrop = CoinDrop;
	InstType = OrcType;
}

Orc::~Orc()
{

}



int Orc::GetType()
{
	return InstType;
}

void Orc::PrintType() const
{
	std::cout << "orc";
}

int Orc::Attack()
{
	std::cout << "The orc hits for:" << club;
	return club;
}

void Orc::Print() const
{
	std::cout << "This orcs club deals " << club << " damage and he has " << MaxHealth << "HP" << std::endl;
}


int Orc::Defend(int DmgTaken)
{
	CurrentHealth -= DmgTaken;
	if (CurrentHealth <= 0)
	{
		std::cout << std::endl << "The orc died " << std::endl;
		return 0;
	}
	std::cout << " and the orcs hp now is " << CurrentHealth << std::endl;
	return CurrentHealth;
}

int Orc::GetCoins()
{
	return CoinDrop;
}

File: Java version/Orc.java
public class Orc extends Entity{

    private int Club;
	private int CoinDrop;


    public Orc(int Club, int CoinDrop, int MaxHealth) {
        super(Type.Orc, MaxHealth);
        this.Club = Club;
        this.CoinDrop = CoinDrop;
    }


    //text output methods
    public void PrintType() {
        System.out.println("orc");
    }


    public void Print() {
        System.out.println("This orc's club deals " + Club + 
        " damage and he has " + MaxHealth + " maximum HP");
    }

    //getters and setters
    public Type GetType() {
        return InstType;
    }


    public int GetCoins() {
        return CoinDrop;
    }

    public void AddCoins(int Amount) {
        return;
    }

    
    //gameplay methods
    public int Attack() {
        System.out.println("The orc hits for: " + Club);
        return Club;
    }


    public int Defend(int Dmgtaken) {
        CurrentHealth -= Dmgtaken;
        if (CurrentHealth <= 0) {
            System.out.println("The orc died ");
            return 0;
        }
        System.out.println(" and the orc's HP is now " + CurrentHealth);
        return CurrentHealth;
    }


    public void UseItem(int ItemID, int Effect) {

    }

}
